### PR TITLE
Feature favor l1 d keyword priority

### PR DESCRIPTION
**Pull request by jin120811**

### CODE REVIEW

**Review by linjuanZ — APPROVED** [review]
Merge this after timing assessment.

**Comment by linjuanZ** [comment]
Merge this after FMA bug is solved, or merge this into new backend.

**Review by Ivyfeather — APPROVED** [review]

**Pull request merged**
